src/main/java/org/opensearch/performanceanalyzer/writer/EventLogQueueProcessor.java

```diff
@@ -97,7 +114,19 @@ public void scheduleExecutor() {
 
     // This executes every purgePeriodicityMillis interval.
     public void purgeQueueAndPersist() {
-        // Return if the writer is not enabled.
+        counter += 1;
```

Comment:
Instead of using counter here why don't we track against the wall clock time ? 
What I mean by that is - user provided input can be obtained form `getMetricsDeletionInterval()`. 
Why don't we keep a member variable called `lastCleanupTimeMillis` or something and we invoke file deletion when `System.currentTimeMillis() - lastCleanupTimeMillis > getMetricsDeletionInterval()` ?

Reply:
Makes sense, moved this away from counter to use the current bucket, last cleanup bucket and the metric deletion interval.